### PR TITLE
remove create method with unused parameter.

### PR DESCRIPTION
**Pull request by RobAltena**
## What changes were proposed in this pull request?
Removed one unused/untested Nd4j.create method.
Removed one  Nd4j.create method with an unused parameter, but used in a test.
Updated this test to use a different create method.

## How was this patch tested?
Test copied to standalone test project. Verified the replacing create method creates the identical array. Verified changed test passes before/after. (even though it is flagged `@ignore`).

As I am removing more unused methods: Before every push unit tests are run on a local build. Results are the same as those on a clean build of master.

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 1]
```diff
@@ -5064,122 +4990,29 @@ public static INDArray create(double[] data, int[] shape, long offset) {
...
 
-
-    // Todo: Only used from tests.
     /**
      * @param data
      * @param indices
      * @param shape
      * @return a INDArray
      * */
     public static INDArray createSparseCOO(double[] data, int[][] indices, long[] shape) {
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: Let's get rid of alll the createSparse methods.
Yes we'll have sparse support in the future, but until then we shouldn't have non-functional methods in the API.
As for where it's used (presumably in the tests) - let's comment those out?

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 2]
```diff
@@ -4983,28 +4952,6 @@ public static INDArray createUninitialized(long... shape) {
...
-    }
-
-    // TODO: unused method.
-    /**
-     * Cretes uninitialized INDArray detached from any (if any) workspace
-     *
-     * @param shape
-     * @return
-     */
-    public static INDArray createUninitializedDetached(long... shape) {
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: For these 'createUninitializedDetached': let's clean this up.

There's basically two methods we really need:
`createUninitializedDetached(DataType, long... shape)`.
`createUninitializedDetached(DataType, char, long... shape)`.
I'm in favor of removing all other methods here.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java (hunk 1 above)
Comment (RobAltena) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: Will be done in a separate small PR.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java (hunk 2 above)
Comment (RobAltena) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: Will be done in a separate small PR.

**Comment by RobAltena** [comment]
See the review conversations: The requested changes will be done as new PRs.

**Pull request merged**